dynamiqs/time_qarray.py

```diff
@@ -291,6 +302,11 @@ def ndim(self) -> int:
     def ndiags(self) -> int:
         pass
 
+    @property
+    @abstractmethod
+    def offsets(self) -> tuple[int, ...]:
+        pass
```

Comment:
This is ill-defined for generic QArrays, since DenseQArrays won't have this property. Also, it looks quite unrelated to the PR and you don't use it, so I'd be in favor of removing this here and in children classes.